unity/unity-embed-host/CoreCLRHost.cs

```diff
@@ -154,14 +289,56 @@ public static int array_length([NativeCallbackType("MonoArray*")] IntPtr array)
         [NativeCallbackType("MonoClass*")] IntPtr klass)
         => FormatterServices.GetUninitializedObject(klass.TypeFromHandleIntPtr()).ToNativeRepresentation();
 
+    [return: NativeCallbackType("MonoException*")]
+    public static IntPtr exception_from_name_msg(
+        [NativeCallbackType("MonoImage*")] IntPtr image,
+        [NativeCallbackType("const char*")] sbyte* name_space,
+        [NativeCallbackType("const char*")] sbyte* name,
+        [NativeCallbackType("const char*")] sbyte* msg)
+    {
+        Type type = class_from_name(image, name_space, name, false).TypeFromHandleIntPtr();
+        return Activator.CreateInstance(type, new string(msg)).ToNativeRepresentation();
+    }
+
     [return: NativeCallbackType("MonoObject*")]
     public static IntPtr type_get_object(
         [NoManagedWrapper] [NativeCallbackType("MonoDomain*")]
         IntPtr domain,
         [NativeCallbackType("MonoType*")] IntPtr type)
         => type.TypeFromHandleIntPtr().ToNativeRepresentation();
 
-    static StringPtr StringToPtr(string s)
+    public static bool class_has_attribute(
+        [NativeCallbackType("MonoClass*")] IntPtr klass,
+        [NativeCallbackType("MonoClass*")] IntPtr attr_klass)
+    {
+        return Attribute.GetCustomAttribute(klass.TypeFromHandleIntPtr(), attr_klass.TypeFromHandleIntPtr()) != null;
+    }
+
+    public static bool assembly_has_attribute(
+        [NativeCallbackType("MonoAssembly*")] IntPtr assembly,
+        [NativeCallbackType("MonoClass*")] IntPtr attr_klass)
+    {
+        return Attribute.GetCustomAttribute(GetAssemblyForHandle(assembly), attr_klass.TypeFromHandleIntPtr()) != null;
+    }
+
+    public static bool method_has_attribute(
+        [NativeCallbackType("MonoMethod*")] IntPtr method,
+        [NativeCallbackType("MonoClass*")] IntPtr attr_class)
+    {
+        MethodInfo mi = Unsafe.As<MethodInfo>(method);
```

Comment:
This is a method handle I believe, not a `MethodInfo` object. You'll want to go from IntPtr -> RuntimeMethodHandle -> MethodInfo.

Reply:
I actually need to add a test for this now that you mention it. I'll do that as well